BoneReconstructionPlanner/BoneReconstructionPlanner.py

```diff
@@ -553,120 +597,161 @@ def process(self,fibulaLine,mandibularCurve,planeList,initialSpace,betweenSpace)
+      
+      mandibleAxisX = [0,0,0]
+      vtk.vtkMath.Cross(mandiblePlane0Y, mandibleAxisZ, mandibleAxisX)
+      mandibleAxisX = mandibleAxisX/np.linalg.norm(mandibleAxisX)
+      mandibleAxisY = [0,0,0]
+      vtk.vtkMath.Cross(mandibleAxisZ, mandibleAxisX, mandibleAxisY)
+      mandibleAxisY = mandibleAxisY/np.linalg.norm(mandibleAxisY)
 
       #Start transformations
       rotAxis = [0,0,0]
-      vtk.vtkMath.Cross(dVersor, lineDirectionVersor, rotAxis)
+      vtk.vtkMath.Cross(mandibleAxisZ, fibulaZ, rotAxis)
       rotAxis = rotAxis/np.linalg.norm(rotAxis)
-      angleRad = vtk.vtkMath.AngleBetweenVectors(dVersor, lineDirectionVersor)
+      angleRad = vtk.vtkMath.AngleBetweenVectors(mandibleAxisZ, fibulaZ)
       angleDeg = vtk.vtkMath.DegreesFromRadians(angleRad)
 
-      #this versor is created to check if rotAxis is okey or should be opposite
-      v1l = [0,0,0]
-      q = [angleRad,rotAxis[0],rotAxis[1],rotAxis[2]]
-      vtk.vtkMath.RotateVectorByWXYZ(dVersor,q,v1l)
+      #this vector is created to check if rotAxis is okey or should be opposite
+      rotatedMandibleAxisZ = [0,0,0]
+      rotation = [angleRad,rotAxis[0],rotAxis[1],rotAxis[2]]
+      vtk.vtkMath.RotateVectorByWXYZ(mandibleAxisZ,rotation,rotatedMandibleAxisZ)
+
+      rotatedMandibleAxisX = [0,0,0]
+      vtk.vtkMath.RotateVectorByWXYZ(mandibleAxisX,rotation,rotatedMandibleAxisX)
 
-      difference = np.linalg.norm(lineDirectionVersor-v1l)
+      difference = np.linalg.norm(fibulaZ-rotatedMandibleAxisZ)
       if (difference>0.01):
         rotAxis = [-rotAxis[0],-rotAxis[1],-rotAxis[2]]
-        q = [angleRad,rotAxis[0],rotAxis[1],rotAxis[2]]
-        vtk.vtkMath.RotateVectorByWXYZ(plane1Normal,q,v1l)
+        rotation = [angleRad,rotAxis[0],rotAxis[1],rotAxis[2]]
+        vtk.vtkMath.RotateVectorByWXYZ(mandibleAxisZ,rotation,rotatedMandibleAxisZ)
+        vtk.vtkMath.RotateVectorByWXYZ(mandibleAxisX,rotation,rotatedMandibleAxisX)
       self.rotTransformParameters.append([rotAxis,angleDeg])
 
-      transformFidA = slicer.vtkMRMLLinearTransformNode()
-      transformFidA.SetName("Mandible2Fibula Transform%d_A" % i)
-      slicer.mrmlScene.AddNode(transformFidA)
-      transformFidB = slicer.vtkMRMLLinearTransformNode()
-      transformFidB.SetName("Mandible2Fibula Transform%d_B" % i)
-      slicer.mrmlScene.AddNode(transformFidB)
+      #Start transformations
+      rotAxis2 = fibulaZ
+      rotAxis2 = rotAxis2/np.linalg.norm(rotAxis2)
+      angleRad2 = vtk.vtkMath.AngleBetweenVectors(rotatedMandibleAxisX, fibulaX)
+      angleDeg2 = vtk.vtkMath.DegreesFromRadians(angleRad2)
+
+      #this vector is created to check if rotAxis is okey or should be opposite
+      doublyRotatedMandibleAxisX = [0,0,0]
+      rotation2 = [angleRad2,rotAxis2[0],rotAxis2[1],rotAxis2[2]]
+      vtk.vtkMath.RotateVectorByWXYZ(rotatedMandibleAxisX,rotation2,doublyRotatedMandibleAxisX)

```

Comment:
Similarly to above, getting a rotation from somewhere and then applying it to some rotation around an axis is just too messy. Always rotate around axis of a well-defined coordinate system.